test/chain-test.js

```diff
@@ -815,6 +815,34 @@ describe('Chain', function() {
       'bad-txns-txouttotal-toolarge');
   });
 
+  it('should fail to connect total fee toolarge', async () => {
+    const job = await cpu.createJob();
+
+    Selector.MAX_FEE = 50 * consensus.COIN;
+
+    const outputs = [{ address: wallet.getAddress(), value: 0 }];
+    const tx1 = await wallet.send({
+      outputs: outputs,
+      hardFee: Selector.MAX_FEE
+    });
+    job.pushTX(tx1.toTX());
+
+    const tx2 = await wallet.send({
+      outputs: outputs,
+      hardFee: Selector.MAX_FEE
+    });
+    job.pushTX(tx2.toTX());
+
+    consensus.MAX_MONEY = tx1.getFee() + tx2.getFee() - 1;
+
+    job.refresh();
+    assert.strictEqual(await mineBlock(job),
+      'bad-txns-accumulated-fee-outofrange');
+
+    consensus.MAX_MONEY = 21000000 * consensus.COIN;
```

Comment:
If the assertion fails, the consensus value isn't restored to the previous value?

Reply:
Hmm, would a `try... finally` work?

Reply:
Yeah, that would likely work.

Reply:
Looks like it does...

Reply:
Thanks, done.